### PR TITLE
Transactions List (account effects)

### PR DESCRIPTION
**Pull request by Daniel-Wang**
- Add view holders for recycler view
- Added a custom ArrayList type that specifically handles the contents of a mixed type array for the RV
- Polling every 5 seconds to refresh the data (auto updates it 🎉)
- Error handling: If there is no network connection, a toast appears at the bottom say "There was a problem connecting to the network. Attempting to reconnect"
- Colored dots indicate receiving, sending money (green, red) and grey for neutral operations.

Known Issues to be polished:
- The trades view holder is not completed, once I work on "switching assets" I will complete this part, since it relies on showing the amount of sold or bought for the "current" asset the user is on.
- `KeyPair` dependency is resolved and awaiting a [release](https://github.com/stellar/java-stellar-sdk/releases), this will significantly improve the speed of the first request for data which is currently taking way too long to load (5-8 seconds) because of the time it takes to generate the `KeyPair`. I have stashed code to fix this but since we're releasing in 2-3 weeks I figured we'll just wait for the release to happen.
- Possible rounding of decimal places to show 4 instead of 7 which is returned by API

Preview: 
screenshot_20180920-165557

### CODE REVIEW

**Comment by satbam** [comment]
@Daniel-Wang This is not an official review but just want to say it looks fantastic. Nice work!

File: app/src/main/java/blockeq/com/stellarwallet/adapters/WalletRecyclerViewAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+        viewHolder.amount!!.text = truncateDecimalPlaces(trade.boughtAmount) //TODO: Equals which ever asset is the selected one? Adding braces depending on that
+        viewHolder.date!!.text = getFormattedDate(trade.createdAt)
+
+        viewHolder.transactionType!!.text = "Trade " + trade.soldAsset + " for " + trade.boughtAsset
+    }
+
+    //endregion
+
+    private fun getFormattedDate(str: String):String {
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/adapters/WalletRecyclerViewAdapter.kt]: Recycler view looks good. I would move `getFormattedDate` and `truncateDecimalPlaces` to utilities. Under trading we'll probably re-use these helpers.

File: app/src/main/java/blockeq/com/stellarwallet/adapters/WalletRecyclerViewAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+    interface OnAssetDropdownListener {
+        fun onAssetDropdownClicked(view : View, position : Int)
+    }
+
+    interface OnLearnMoreButtonListener {
+        fun onLearnMoreButtonClicked(view : View, position : Int)
+    }
+
+    enum class TransactionViewType(val value :Int) {
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/adapters/WalletRecyclerViewAdapter.kt]: 👌 very nice and readable

File: app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt [hunk 3]
```diff
@@ -50,47 +60,116 @@ class WalletFragment : BaseFragment(), OnLoadAccount {
...
 
     private fun startPollingAccount() {
         runnableCode = object : Runnable {
             override fun run() {
 
-                Horizon.Companion.LoadAccountTask(this@WalletFragment)
-                        .execute(WalletApplication.session!!.keyPair)
+                if (WalletApplication.session != null && isNetworkAvailable()) {
+
+                    Horizon.Companion.LoadAccountTask(this@WalletFragment)
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt]: Just as a note here - we'll poll only if the user account is not created on Stellar. To know whether account is created, the horizon will send you `Account doesn't exist or not found` (or something similar).

File: app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt [hunk 4]
```diff
@@ -50,47 +60,116 @@ class WalletFragment : BaseFragment(), OnLoadAccount {
...
+        handler.post(runnableCode)
     }
 
     private fun endPollingAccount() {
         handler.removeCallbacks(runnableCode)
     }
 
-    override fun onLoadAccount(result: AccountResponse?) {
-        loadBalance()
+    private fun isNetworkAvailable(): Boolean {
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt]: Great thinking here. We can move this off to utilities as well.

File: app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt [hunk 5]
```diff
@@ -50,47 +60,116 @@ class WalletFragment : BaseFragment(), OnLoadAccount {
...
 
     //region API Polling
 
     private fun startPollingAccount() {
         runnableCode = object : Runnable {
             override fun run() {
 
-                Horizon.Companion.LoadAccountTask(this@WalletFragment)
-                        .execute(WalletApplication.session!!.keyPair)
+                if (WalletApplication.session != null && isNetworkAvailable()) {
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt]: Nice call on network check. However, `isNetworkAvailable` would be better off in the `Horizon` services. If not available, a callback can send an error message saying `Network not available` and we can handle in UI by showing a snack bar.

File: app/src/main/res/drawable/colored_dot.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (MeghaBambra) [inline app/src/main/res/drawable/colored_dot.xml]: Nice!

File: app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt [hunk 7]
```diff
@@ -50,47 +60,116 @@ class WalletFragment : BaseFragment(), OnLoadAccount {
...
+    //region User Interface
+
     private fun setupUI() {
+        bindAdapter()
         loadBalance()
     }
 
-    private fun loadBalance() {
+    private fun bindAdapter() {
+        recyclerViewArrayList = WalletHeterogenousArray(TotalBalance(loadBalance()),
```
Comment (MeghaBambra) [inline app/src/main/java/blockeq/com/stellarwallet/fragments/WalletFragment.kt]: Nice that the data organization in the array is off loaded and not in the activity 👍

**Comment by MeghaBambra** [comment]
@Daniel-Wang lots of great work in this PR! We can talk more about the Stellar account not created tomorrow but doesn't have to be done in this PR - we can merge it in with other minor changes.

**Pull request merged**